timely_beliefs/beliefs/classes.py

```diff
@@ -465,6 +465,14 @@ def __init__(  # noqa: C901
     ):  # noqa: C901 todo: refactor, e.g. by detecting initialization method
         """Initialise a multi-index DataFrame with beliefs about a unique sensor."""
 
+        # Initialized with a BeliefsSeries or BeliefsDataFrame
+        if len(args) > 0 and isinstance(args[0], (BeliefsSeries, BeliefsDataFrame)):
+            super().__init__(*args, **kwargs)
+            assign_sensor_and_event_resolution(
```

Comment:
It seems to me that we only call `pd.DataFrame.__init__` here, which won't understand an argument like `beliefs` or `event_start`? These are read in lines 476ff which are now never run for this case.

Reply:
That's correct. Those arguments should only play a role when initializing from a non-(BeliefsSeries or BeliefsDataFrame). That is, all information about events, sensors and sources is already contained in the BeliefsSeries or BeliefsDataFrame.

Reply:
Okay, but how is `args[0]` being preserved within `pd.DataFrame.__init__` Does that do a `self = args[0]` assignment?

Reply:
The init methods of `pd.Series` and `pd.DataFrame` both try to extract a [block manager](https://uwekorn.com/2020/05/24/the-one-pandas-internal.html), but in different ways. What are you worried about? That the init method can still change the input in this case? Do you propose to add a copy statement here, too?

Reply:
The changing of input was not my worry, but is also interesting to check.

I didn't understand what __init__ is doing with args[0] but apparantly you are confident that the important columns (datetime and event_value) are preserved in the way we expect in a BeliefsDataFrame.

Reply:
Yes, that I am confident of. I'll add a test for initializing a BeliefsDataFrame from a BeliefsSeres or BeliefsDataFrame.